packages/generator/scripts/sync-packages.ts

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env node
+
+import fs from 'fs';
+
+import currentDirectoryData from '../directory-data.json';
+import { downloadFontAssets } from '../downloadFontAssets';
+import { generateImages } from '../generateImages';
+import getGoogleFontsApiKey from '../google-fonts-api-key';
+import {
+  generateDevPackage,
+  generateFontPackage,
+  generateFontDirectoryPackage,
+  generateRootReadme,
+  generateGalleryFile,
+} from '../shared';
+import { FontItem } from '../types';
+const currentDirectoryItems = currentDirectoryData.items as FontItem[];
+
+async function syncPackages() {
+  // fetch the latest directory data from the Google Fonts API
+  const googleFontsApiKey = getGoogleFontsApiKey();
+  const url = `https://www.googleapis.com/webfonts/v1/webfonts?key=${googleFontsApiKey}&prettyPrint=false&sort=date`;
+  const response = await fetch(url);
+  const fetchedDirectoryData = (await response.json()) as { items: FontItem[] };
+  const fetchedDirectoryDataItems = fetchedDirectoryData.items;
+
+  // compare the fetched directory data with the current directory data
+  const deletedPackages = currentDirectoryItems.filter(
+    (item) => !fetchedDirectoryDataItems.find((p) => p.family === item.family)
+  );
+  const newPackages = fetchedDirectoryDataItems.filter(
+    (item) => !currentDirectoryItems.find((p) => p.family === item.family)
+  );
+  const changedPackages = fetchedDirectoryDataItems.filter((item) => {
+    const currentPackage = currentDirectoryItems.find((p) => p.family === item.family);
+    return currentPackage && currentPackage?.lastModified !== item.lastModified;
+  });
+
+  if (!deletedPackages.length && !newPackages.length && !changedPackages.length) {
+    console.log('No changes to the directory data.');
+    return;
+  }
+
+  if (deletedPackages.length) {
+    console.log(
+      `\nFound ${deletedPackages.length} deleted package${deletedPackages.length === 1 ? '' : 's'}`
+    );
+    // TODO: Do we delete the packages?
+  }
```

Comment:
Great! I'll create a follow-up PR to handle the deleted packages

Reply:
Perfect, thank you!